readme.md

```diff
@@ -783,6 +783,37 @@ It's important to note, that by design this extension will NOT use the [dekorate
 
 For more information plese check: the extension [docs](https://quarkus.io/guides/ap4k).
 
+#### Thorntail
+
+With Thorntail, it is recommended to add a dependency on one of the provided starters:
+
+```xml
+<dependency>
+  <groupId>io.dekorate</groupId>
+  <artifactId>kubernetes-thorntail-starter</artifactId>
+  <version>${project.version}</version>
+  <scope>provided</scope>
```

Comment:
So this is something I wanted to ask about, actually. If I left out `<scope>provided</scope>`, I was getting weird failures from the Thorntail Maven plugin, because it tried to find `com.sun:tools:jar:${sun.tools.version}:system`. Here's the relevant part of `mvn dependency:tree`:

```
io.dekorate:thorntail-on-kubernetes-example:war:0.8-SNAPSHOT
+- io.dekorate:kubernetes-thorntail-starter:jar:0.8-SNAPSHOT:provided
|  +- io.dekorate:option-annotations:jar:annotations:0.8-SNAPSHOT:provided
|  |  \- io.dekorate:dekorate-core:jar:0.8-SNAPSHOT:provided
|  |     +- io.dekorate:dekorate-dependencies:jar:0.8-SNAPSHOT:provided
|  |     \- io.sundr:transform-annotations:jar:0.19.1:provided
|  +- io.dekorate:kubernetes-annotations:jar:annotations:0.8-SNAPSHOT:provided
|  +- io.dekorate:servicecatalog-annotations:jar:annotations:0.8-SNAPSHOT:provided
|  |  \- io.sundr:builder-annotations:jar:0.19.1:provided
|  |     +- io.sundr:sundr-core:jar:0.19.1:provided
|  |     +- io.sundr:sundr-codegen:jar:0.19.1:provided
|  |     +- io.sundr:resourcecify-annotations:jar:0.19.1:provided
|  |     \- com.sun:tools:jar:${sun.tools.version}:system
|  +- io.dekorate:option-annotations:jar:processors:0.8-SNAPSHOT:provided
|  +- io.dekorate:kubernetes-annotations:jar:processors:0.8-SNAPSHOT:provided
|  +- io.dekorate:servicecatalog-annotations:jar:processors:0.8-SNAPSHOT:provided
|  \- io.dekorate:dekorate-thorntail:jar:0.8-SNAPSHOT:provided
|     \- io.dekorate:openshift-annotations:jar:0.8-SNAPSHOT:provided
```

I'm not quite sure why `io.sundr:builder-annotations` comes through `io.dekorate:servicecatalog-annotations`. I would expect that everything `io.sundr` would be `<scope>provided</scope>` in Dekorate.

In any case, I think marking Dekorate dependencies in end user's applications as `<scope>provided</scope>` is always a good idea, as Dekorate does everything at build time (unfortunately Maven doesn't have anything like `<scope>compile-only</scope>).

Reply:
Yeah, I am always baffled by this.... I really don't understand why it doesn't...

Reply:
sundrio dependencies should be compile only and optional.

Reply:
Where do I file a bug then? :-)

Reply:
Created https://github.com/dekorateio/dekorate/issues/340

Reply:
Thanks! I'll remember for future :-)